piff/stats.py

```diff
@@ -0,0 +1,364 @@
+
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: stats
+
+"""
+
+from __future__ import print_function
+import numpy as np
+
+def process_stats(config, logger):
+    """Parse the stats field of the config dict.
+
+    :param config:      The configuration dict.
+    :param logger:      A logger object for logging debug info.
+
+    :returns: an stats instance
+    """
+    import piff
+
+    if 'stats' not in config:
+        raise ValueError("config dict has no stats field")
+    config_stats_list = config['stats']
+
+    stats = []
+    for config_stats in config_stats_list:
+
+        if 'type' not in config_stats:
+            raise ValueError("config['stats'] has no type field")
+
+        # Get the class to use for the stats
+        # Not sure if this is what we'll always want, but it would be simple if we can make it work.
+        stats_class = eval('piff.' + config_stats.pop('type'))
```

Comment:
The logic on that was that other than the 'type' parameter in the config dict, the rest of the parameters there would normally be the construction kwargs for the builder.  So the `parseKwargs` function in each of the builder base classes just returns the rest of the config dict (although it can be overridden if there is more work to be done for a particular class).

Chris didn't use this paradigm for the stats classes, so maybe here it doesn't make sense.

As for whether that's a problem for repeated runs, we should probably just make a deep copy at the start of each run.  Then we can mess with the config dict as much as we want along the way without screwing things up for other runs.